judge/forms.py

```diff
@@ -69,13 +69,19 @@ def __init__(self, *args, **kwargs):
 
 class ProblemSubmitForm(ModelForm):
     source = CharField(max_length=65536, widget=AceWidget(theme='twilight', no_ace_media=True))
+    judge = ChoiceField(choices=(), widget=forms.HiddenInput(), required=False)
 
     def __init__(self, *args, **kwargs):
+        judge_queryset = kwargs.pop('judge_queryset', ())
```

Comment:
Logic before calling super is pretty yikes. I think you can manually specify it in `__init__`, in which case it won't be duplicated in `kwargs`?